sympy/parsing/latex/lark/grammar/latex.lark

```diff
@@ -110,27 +110,82 @@ UPRIGHT_DIFFERENTIAL_SYMBOL: "\\text{d}" | "\\mathrm{d}"
 DIFFERENTIAL_SYMBOL: "d" | UPRIGHT_DIFFERENTIAL_SYMBOL
 
 // disallow "d" as a variable name because we want to parse "d" as a differential symbol.
-SYMBOL: /[a-zA-Z]/
-BASIC_SUBSCRIPTED_SYMBOL: /([a-zA-Z])_(([A-Za-z0-9]|[a-zA-Z]+)|\{([A-Za-z0-9]|[a-zA-Z]+)\})/
-SYMBOL_WITH_GREEK_SUBSCRIPT: /([a-zA-Z])_/ GREEK_SYMBOL | /([a-zA-Z])_/ L_BRACE GREEK_SYMBOL R_BRACE
+SYMBOL: /[a-zA-Z]'*/
+GREEK_SYMBOL_WITH_PRIMES: GREEK_SYMBOL "'"*
+LATIN_SYMBOL_WITH_LATIN_SUBSCRIPT: /([a-zA-Z]'*)_(([A-Za-z0-9]|[a-zA-Z]+)|\{([A-Za-z0-9]|[a-zA-Z]+'*)\})/
+LATIN_SYMBOL_WITH_GREEK_SUBSCRIPT: /([a-zA-Z]'*)_/ GREEK_SYMBOL | /([a-zA-Z]'*)_/ L_BRACE GREEK_SYMBOL_WITH_PRIMES R_BRACE
 // best to define the variant with braces like that instead of shoving it all into one case like in
 // /([a-zA-Z])_/ L_BRACE? GREEK_SYMBOL R_BRACE? because then we can easily error out on input like
 // r"h_{\theta"
-GREEK_SUBSCRIPTED_SYMBOL: GREEK_SYMBOL /_(([A-Za-z0-9]|[a-zA-Z]+)|\{([A-Za-z0-9]|[a-zA-Z]+)\})/
+GREEK_SYMBOL_WITH_LATIN_SUBSCRIPT: GREEK_SYMBOL_WITH_PRIMES /_(([A-Za-z0-9]|[a-zA-Z]+)|\{([A-Za-z0-9]|[a-zA-Z]+'*)\})/
+GREEK_SYMBOL_WITH_GREEK_SUBSCRIPT: GREEK_SYMBOL_WITH_PRIMES /_/ (GREEK_SYMBOL | L_BRACE GREEK_SYMBOL_WITH_PRIMES R_BRACE)
+MULTI_LETTER_SYMBOL: /[a-zA-Z]+(\s+[a-zA-Z]+)*'*/
 
 %import common.DIGIT -> DIGIT
 
+CMD_PRIME: "\\prime"
+CMD_ASTERISK: "\\ast"
+
+PRIMES: "'"+
+STARS: "*"+
+PRIMES_VIA_CMD: CMD_PRIME+
+STARS_VIA_CMD: CMD_ASTERISK+
+
+CMD_IMAGINARY_UNIT: "\\imaginaryunit"
```

Comment:
@sylee957 Now I am on my computer; detailed response follows.

Here is my use case: let's say I typeset two literal matrices in LaTeX in my emacs buffer; these matrices are simply placed one after the other (so that it looks like matrix multiplication when complied). These matrices have complex entries.

Then I want to be able to select the region containing these matrices, and then invoke an elisp function (via some convenient keybinding, for example), which does all of the following steps in sequence without manual intervention:

1. call `parse_latex()` to convert the LaTeX to sympy,
2. multiply the two matrices using sympy,
3. convert the result back to LaTeX using `latex()`, and finally
4. insert ` = ` followed by the LaTeX from step 3 into the emacs buffer after point.

This allows for high productivity when I am working on a paper.

Of course, if any user 1. does not like `\imaginaryunit`, 2. does not wish to define their own macro for the imaginary unit, and 3. wishes to continue using `i`, they **still have the choice** to do it and perform post-processing like you mentioned. This PR **gives one the option** to use a LaTeX macro (which would simply expand to the letter `i` or `\iota` in LaTeX via `\newcommand`, for example) which sympy understands, allowing for high productivity without the need for manual post-processing.

If you do not like this macro-based approach, then please suggest a clear alternative way to perform my automatic workflow listed above. Also, suggest similar alternatives for matrix trace and adjugate.